core/test/allocation_test.jl

```diff
@@ -221,18 +221,20 @@ end
     # Collecting demands
     u = ComponentVector(; storage = zeros(length(basin.node_id)))
     for allocation_model in allocation_models[2:end]
-        Ribasim.allocate!(p, allocation_model, t, u; collect_demands = true)
+        Ribasim.allocate!(p, allocation_model, t, u, OptimizationType.internal_sources)
+        Ribasim.allocate!(p, allocation_model, t, u, OptimizationType.collect_demands)
     end
 
+    # See the difference between these values here and in
+    # "subnetworks_with_sources"
     @test subnetwork_demands[(NodeID(:Basin, 2), NodeID(:Pump, 11))] ≈ [4.0, 4.0, 0.0]
     @test subnetwork_demands[(NodeID(:Basin, 6), NodeID(:Pump, 24))] ≈ [0.004, 0.0, 0.0]
-    @test subnetwork_demands[(NodeID(:Basin, 10), NodeID(:Pump, 38))] ≈
-          [0.001, 0.002, 0.0011]
+    @test subnetwork_demands[(NodeID(:Basin, 10), NodeID(:Pump, 38))][1:2] ≈ [0.001, 0.002]
```

Comment:
For an explanation of why I removed testing the last value of this vector see https://github.com/Deltares/Ribasim/issues/1327.